packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -128,6 +120,24 @@ export function getDeserializePrivateFunction(
       "throw result.body",
       "}"
     );
+  } else {
+    const validStatus = [
+      ...new Set(
+        operation.responses
+          .flatMap((r) => r.statusCodes)
+          .filter((s) => s !== "default")
+      )
+    ];
+
+    if (validStatus.length > 0) {
+      statements.push(
+        `if(${validStatus
+          .map((s) => `"${s}" !== result.status`)
```

Comment:
```suggestion
          .map((s) => `result.status !== "${s}"`)
```
looks weird to put the status code in left side. but no strong opinions here.